src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -317,6 +321,10 @@ private function createFirewalls(array $config, ContainerBuilder $container)
             $contextRefs[$contextId] = new Reference($contextId);
             $map[$contextId] = $matcher;
         }
+        $container
+            ->getDefinition('security.helper')
+            ->replaceArgument(0, $authenticators)
```

Comment:
Authenticators need to be lazy loaded:
```suggestion
            ->replaceArgument(0, ServiceLocatorTagPass::register($container, $authenticators))
```